### PR TITLE
Add unit tests for GcpBodyFieldSanitizer in google cloud providers

### PR DESCRIPTION
**Pull request by rootcss**
Adds unit tests for GcpBodyFieldSanitizer in google cloud providers, as mentioned as missing unit tests in issue https://github.com/apache/airflow/issues/8278

### CODE REVIEW

File: tests/providers/google/cloud/utils/test_field_sanitizer.py [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+
+        self.assertEqual({}, body)
+
+    def test_sanitize_should_not_remove_fields_for_incorrect_specification(self):
+        actual_body = [
+            {"kind": "compute#instanceTemplate", "name": "instance"},
+            {"kind": "compute#instanceTemplate1", "name": "instance1"},
+            {"kind": "compute#instanceTemplate2", "name": "instance2"},
+        ]
+        body = actual_body
```
Comment (mik-laj) [inline tests/providers/google/cloud/utils/test_field_sanitizer.py]:
```suggestion
        body = deepcopy(actual_body)
```
We want to have assertions on nested objects, so we need to have copies or the behavior may be different than expected.

Comment (rootcss) [reply tests/providers/google/cloud/utils/test_field_sanitizer.py]: correc 👍  dicts share the reference, hence require the deepcopy.

Comment (rootcss) [reply tests/providers/google/cloud/utils/test_field_sanitizer.py]: Can you please look at this? https://github.com/apache/airflow/issues/8278#issuecomment-663153829 @mik-laj

**Comment by rootcss** [comment]
CI Build: Quarantined:Pg9.6,Py3.6 failed abruptly. Is it a known issue?
https://github.com/apache/airflow/pull/9996/checks?check_run_id=910077269

**Review by mik-laj — APPROVED** [review]

**Pull request merged**